src/core/operators/file_processing_op.py

```
@@ -0,0 +1,344 @@
+        scenes = sm.get_scene_list()
+
+        frames: List[Image.Image] = []
+        if scenes:
+            for i in np.linspace(0, len(scenes) - 1, min(len(scenes), desired), dtype=int):
+                frames.append(_capture_screenshot(video_path, scenes[i][0].get_seconds()))
+        
+        while len(frames) < desired and duration:
+            t = len(frames) * frame_interval
+            try:
+                frames.append(_capture_screenshot(video_path, t))
+            except Exception as e:  # short file edge‑case
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)